src/main/java/frc/robot/subsystems/ArmSubsystem.java

```diff
@@ -0,0 +1,297 @@
+  private RelativeEncoder m_bottomEncoder;
+  private RelativeEncoder m_topEncoder;
+  public final double kP = 0.2;
+  public final double kI = 0;
+  public final double kD = 0; 
+  public final double kIz = 0;
+  public final double kFF = 0.4;
+  public final double kMaxOutput = 0.5;
+  public final double kMinOutput = -0.5;
+
+  // for top arm
+  private final String m_tuningTableTop = "Arm/TopArmTuning";
+  private final String m_dataTableTop = "Arm/TopArmData";
+
+  //for bottom arm
+  private final String m_tuningTableBottom = "Arm/BottomArmTuning";
+  private final String m_dataTableBottom = "Arm/BottomArmData";
+
+  // network table entries for top arm
+  private DoubleEntry m_topArmPSubs;
+  private DoubleEntry m_topArmISubs;
+  private DoubleEntry m_topArmDSubs;
+  private DoubleEntry m_topArmIzSubs;
+  private DoubleEntry m_topArmFFSubs;
+  private DoublePublisher m_topArmPosPub;
+  private DoublePublisher m_topArmSetpointPub;   
+  private DoublePublisher m_topArmVelPub;
+  private DoubleEntry m_topArmVoltsAtHorizontal;
+  private DoublePublisher m_topArmFFTestingVolts;
+
+  // network table entries for bottom arm
+  private DoubleEntry m_bottomArmPSubs;
+  private DoubleEntry m_bottomArmISubs;
+  private DoubleEntry m_bottomArmDSubs;
+  private DoubleEntry m_bottomArmIzSubs;
+  private DoubleEntry m_bottomArmFFSubs;
+  private DoublePublisher m_bottomArmPosPub;
+  private DoublePublisher m_bottomArmSetpointPub;   
+  private DoublePublisher m_bottomArmVelPub;
+  private DoubleEntry m_bottomArmVoltsAtHorizontal;
+  private DoublePublisher m_bottomArmFFTestingVolts;
+
+  public static ArmSubsystem getInstance() {
+    if (instance == null) {
+      SubsystemChecker.subsystemConstructed(SubsystemType.ArmSubsystem);
+      instance = new ArmSubsystem();
+    }
+    return instance;
+}
+
+  /** Creates a new ArmSubsystem. */
+  public ArmSubsystem() {
+    m_topArm = new CANSparkMax(Config.CANID.TOP_ARM_SPARK_CAN_ID, motorType);
+    m_bottomArm = new CANSparkMax(Config.CANID.BOTTOM_ARM_SPARK_CAN_ID, motorType);
+    CANCoderConfiguration config = new CANCoderConfiguration();
+    m_topArm.restoreFactoryDefaults();
+    m_bottomArm.restoreFactoryDefaults();
+    m_topArm.setSmartCurrentLimit(40);
+    m_bottomArm.setSmartCurrentLimit(40);
+    m_topArm.setInverted(false);
+    m_bottomArm.setInverted(false);
+    m_topArm.setIdleMode(IdleMode.kBrake);
+    m_bottomArm.setIdleMode(IdleMode.kBrake);
+    
+
+    // set units of the CANCoder to radians, with velocity being radians per second
+    config.unitString = "rad";
+    config.sensorTimeBase = SensorTimeBase.PerSecond;
+
+    m_absoluteTopArmEncoder = new CANCoder(16);
+    m_absoluteBottomArmEncoder = new CANCoder(17);
+
+    m_absoluteTopArmEncoder.configAllSettings(config);
+    m_absoluteBottomArmEncoder.configAllSettings(config);
+
+    m_pidControllerTopArm = m_topArm.getPIDController();
+    m_pidControllerBottomArm = m_bottomArm.getPIDController();
+    updatePIDSettings();
```

Comment:
updatePIDSettings uses the DoubleSubcriber objects that are defined below this line. So calling updatePIDSettings here should cause a NullPointerException. If you move the call to updatePIDSettings to the bottom of this constructor then that should resolve the NullPointerException.